astronomer/providers/amazon/aws/example_dags/example_batch.py

```diff
@@ -236,6 +238,16 @@ def delete_job_queue_func() -> None:
         raise error
 
 
+def check_dag_status(**kwargs: Any) -> None:
+    """Raises an exception if any of the DAG's tasks failed and as a result marking the DAG failed."""
+    for task_instance in kwargs["dag_run"].get_task_instances():
+        if (
+            task_instance.current_state() != State.SUCCESS
+            and task_instance.task_id != kwargs["task_instance"].task_id
```

Comment:
May I know in what situation would `task_instance.task_id != kwargs["task_instance"].task_id`?

Reply:
For the current task itself that is being executed i.e. dag_final_status task itself.